From 2bfe294a4ed88fa4eb1d612c657637e2d5ceac26 Mon Sep 17 00:00:00 2001
From: Christine Gharzuzi <chrisg@marvell.com>
Date: Wed, 26 Apr 2017 21:37:18 +0300
Subject: [PATCH 0083/1239] arm64: dts: a70x0: add a7040-db device tree for
 setup D

Update setup D of a7040-db according to board topology:
SerDes:
	- lane0: PCIe0
	- lane1: USB0 DEVICE
	- lane2: 10G
	- lane3: SATA1
	- lane4: USB1
	- lane5: PCIe1
AP-MPPs:
	- SDIO, UART
CP-MPPs:
	- RGMII2, USB-DEVICE, NAND, Audio

Change-Id: I090240fa8d77f1598329d3d9ad8e62f8557646f7
Signed-off-by: Christine Gharzuzi <chrisg@marvell.com>
Signed-off-by: Ken Ma <make@marvell.com>
---
 arch/arm/dts/Makefile             |   1 +
 arch/arm/dts/armada-7040-db-D.dts | 185 ++++++++++++++++++++++++++++++
 2 files changed, 186 insertions(+)
 create mode 100644 arch/arm/dts/armada-7040-db-D.dts

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 44d2817afc..32e3149203 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -185,6 +185,7 @@ dtb-$(CONFIG_ARCH_MVEBU) +=			\
 	armada-7040-db.dtb			\
 	armada-7040-db-B.dtb			\
 	armada-7040-db-C.dtb                    \
+	armada-7040-db-D.dtb			\
 	armada-7040-db-nand.dtb			\
 	armada-8040-db.dtb			\
 	armada-8040-db-B.dtb			\
diff --git a/arch/arm/dts/armada-7040-db-D.dts b/arch/arm/dts/armada-7040-db-D.dts
new file mode 100644
index 0000000000..34dc2b6177
--- /dev/null
+++ b/arch/arm/dts/armada-7040-db-D.dts
@@ -0,0 +1,185 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * Copyright (C) 2018 Marvell International Ltd.
+ */
+
+/*
+ * Device Tree file for Marvell Armada 7040 Development board platform
+ * Boot device: SPI NOR, 0x32 (SW3)
+ */
+
+#include "armada-70x0-db.dtsi"
+
+/ {
+	model = "Marvell Armada 7040 Development Board DB-88FA7040 setup(D)";
+	compatible = "marvell,armada7040-db", "marvell,armada7040",
+		     "marvell,armada-ap806-quad", "marvell,armada-ap806";
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	aliases {
+		i2c0 = &cp0_i2c0;
+		spi0 = &cp0_spi1;
+	};
+
+	memory@00000000 {
+		device_type = "memory";
+		reg = <0x0 0x0 0x0 0x80000000>;
+	};
+
+	cp0_sar {
+		boot_mode {
+			default = <0xe>;
+		};
+	};
+};
+
+&ap_pinctl {
+	   /* MPP Bus:
+	    * SDIO  [0-5]
+	    * UART0 [11,19]
+	    */
+		  /* 0 1 2 3 4 5 6 7 8 9 */
+	pin-func = < 1 1 1 1 1 1 0 0 0 0
+		     0 3 0 0 0 0 0 0 0 3 >;
+};
+
+&uart0 {
+	status = "okay";
+};
+
+&cp0_pcie0 {
+	status = "okay";
+};
+
+&cp0_pcie2 {
+	status = "okay";
+};
+
+&cp0_i2c0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&cp0_i2c0_pins>;
+	status = "okay";
+	clock-frequency = <100000>;
+};
+
+&cp0_pinctl {
+		/* MPP Bus:
+		 * AUDIO   [0-5]
+		 * SS_PWDN [12]
+		 * NF_RBn  [13]
+		 * DEV_BUS [14-27]
+		 * SATA1   [28]
+		 * UART0   [29-30]
+		 * MSS_VTT [31]
+		 * SMI	   [32,34]
+		 * XSMI    [35-36]
+		 * I2C	   [37-38]
+		 * RGMII1  [44-55]
+		 * SD	   [56-61]
+		 * GPIO    [62]
+		 */
+		/*   0   1   2   3   4   5   6   7   8   9 */
+	pin-func = < 2   2   2   2   2   2   0   0   0   0
+		     0   0   0   2   0   1   1   1   1   1
+		     1   1   1   1   1   1   1   1   9   0xA
+		     0xA 0   7   0   7   7   7   2   2   0
+		     0   0   0   0   1   1   1   1   1   1
+		     1   1   1   1   1   1    0xE 0xE 0xE 0xE
+		     0xE 0xE 0>;
+};
+
+&cp0_nand {
+	status = "okay";
+};
+
+&cp0_sata0 {
+	status = "okay";
+};
+
+&cp0_usb3_1 {
+	status = "okay";
+};
+
+&cp0_comphy {
+	phy0 {
+		phy-type = <COMPHY_TYPE_PEX0>;
+		phy-speed = <COMPHY_SPEED_5G>;
+	};
+
+	phy1 {
+		phy-type = <COMPHY_TYPE_USB3_DEVICE>;
+		phy-speed = <COMPHY_SPEED_5G>;
+	};
+
+	phy2 {
+		phy-type = <COMPHY_TYPE_SFI>;
+		phy-speed = <COMPHY_SPEED_10_3125G>;
+	};
+
+	phy3 {
+		phy-type = <COMPHY_TYPE_SATA1>;
+		phy-speed = <COMPHY_SPEED_5G>;
+	};
+
+	phy4 {
+		phy-type = <COMPHY_TYPE_USB3_HOST1>;
+		phy-speed = <COMPHY_SPEED_5G>;
+	};
+
+	phy5 {
+		phy-type = <COMPHY_TYPE_PEX2>;
+		phy-speed = <COMPHY_SPEED_5G>;
+	};
+};
+
+&cp0_utmi0 {
+	utmi-port = <UTMI_PHY_TO_USB3_DEVICE0>;
+	status = "okay";
+};
+
+&cp0_utmi1 {
+	status = "okay";
+};
+
+&ap_sdhci0 {
+	status = "okay";
+	bus-width = <4>;
+	no-1-8-v;
+	non-removable;
+};
+
+&cp0_sdhci0 {
+	status = "okay";
+	bus-width = <4>;
+	no-1-8-v;
+	non-removable;
+};
+
+&cp0_mdio {
+	status = "okay";
+	phy0: ethernet-phy@0 {
+		reg = <0>;
+	};
+	phy1: ethernet-phy@1 {
+		reg = <1>;
+	};
+};
+
+&cp0_ethernet {
+	status = "okay";
+};
+
+&cp0_eth1 {
+	status = "okay";
+	phy = <&phy0>;
+	phy-mode = "sgmii";
+};
+
+&cp0_eth2 {
+	status = "okay";
+	phy = <&phy1>;
+	phy-mode = "rgmii-id";
+};
-- 
2.29.0

